flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkTableSource.java

```diff
@@ -611,44 +608,38 @@ public void testFilterPushDown2Literal() {
   public void testSqlParseError() {
     String sqlParseErrorEqual =
         String.format("SELECT * FROM %s WHERE d = CAST('NaN' AS DOUBLE) ", TABLE_NAME);
-    AssertHelpers.assertThrows(
-        "The NaN is not supported by flink now. ",
-        NumberFormatException.class,
-        () -> sql(sqlParseErrorEqual));
+    Assertions.assertThatThrownBy(() -> sql(sqlParseErrorEqual))
+        .as("The NaN is not supported by flink now. ")
+        .isInstanceOf(NumberFormatException.class);
 
     String sqlParseErrorNotEqual =
         String.format("SELECT * FROM %s WHERE d <> CAST('NaN' AS DOUBLE) ", TABLE_NAME);
-    AssertHelpers.assertThrows(
-        "The NaN is not supported by flink now. ",
-        NumberFormatException.class,
-        () -> sql(sqlParseErrorNotEqual));
+    Assertions.assertThatThrownBy(() -> sql(sqlParseErrorNotEqual))
+        .as("The NaN is not supported by flink now. ")
+        .isInstanceOf(NumberFormatException.class);
 
     String sqlParseErrorGT =
         String.format("SELECT * FROM %s WHERE d > CAST('NaN' AS DOUBLE) ", TABLE_NAME);
-    AssertHelpers.assertThrows(
-        "The NaN is not supported by flink now. ",
-        NumberFormatException.class,
-        () -> sql(sqlParseErrorGT));
+    Assertions.assertThatThrownBy(() -> sql(sqlParseErrorGT))
+        .as("The NaN is not supported by flink now. ")
+        .isInstanceOf(NumberFormatException.class);
 
     String sqlParseErrorLT =
         String.format("SELECT * FROM %s WHERE d < CAST('NaN' AS DOUBLE) ", TABLE_NAME);
-    AssertHelpers.assertThrows(
-        "The NaN is not supported by flink now. ",
-        NumberFormatException.class,
-        () -> sql(sqlParseErrorLT));
+    Assertions.assertThatThrownBy(() -> sql(sqlParseErrorLT))
+        .as("The NaN is not supported by flink now. ")
+        .isInstanceOf(NumberFormatException.class);
 
     String sqlParseErrorGTE =
         String.format("SELECT * FROM %s WHERE d >= CAST('NaN' AS DOUBLE) ", TABLE_NAME);
-    AssertHelpers.assertThrows(
-        "The NaN is not supported by flink now. ",
-        NumberFormatException.class,
-        () -> sql(sqlParseErrorGTE));
+    Assertions.assertThatThrownBy(() -> sql(sqlParseErrorGTE))
+        .as("The NaN is not supported by flink now. ")
+        .isInstanceOf(NumberFormatException.class);
 
     String sqlParseErrorLTE =
         String.format("SELECT * FROM %s WHERE d <= CAST('NaN' AS DOUBLE) ", TABLE_NAME);
-    AssertHelpers.assertThrows(
-        "The NaN is not supported by flink now. ",
-        NumberFormatException.class,
-        () -> sql(sqlParseErrorLTE));
+    Assertions.assertThatThrownBy(() -> sql(sqlParseErrorLTE))
+        .as("The NaN is not supported by flink now. ")
+        .isInstanceOf(NumberFormatException.class);
```

Comment:
all of these need to check the underlying error msg. Also `.as()` isn't needed

Reply:
Addressed